src/rdb.c

```diff
@@ -2708,8 +2708,12 @@ void stopLoading(int success) {
 void startSaving(int rdbflags) {
     /* Fire the persistence modules end event. */
     int subevent;
-    if (rdbflags & RDBFLAGS_AOF_PREAMBLE)
-        subevent = REDISMODULE_SUBEVENT_PERSISTENCE_AOF_START;
+    if (rdbflags & RDBFLAGS_AOF_PREAMBLE) {
+        if (getpid() != server.pid)
+            subevent = REDISMODULE_SUBEVENT_PERSISTENCE_AOF_START;
+        else 
+            subevent = REDISMODULE_SUBEVENT_PERSISTENCE_SYNC_AOF_START;
+    }
     else if (getpid()!=server.pid)
```

Comment:
```suggestion
    } else if (getpid()!=server.pid)
```
maybe also convert this to a single if-else chain?